### ISSUE TITLE
Submit issues/commits upstream

### ISSUE THREAD

**Issue by nunoguedelha** [ami-iit/openmct#21]
#### Summary

We track here the submission to upstream repository https://github.com/nasa/openmct of all issues found and respective fixes done so far.

[^1]: submitted by IIT

| Issue # | Pull request # | Commit # | Commit description | Remote issue | IIT[^1] | Remote pull request | IIT[^1] |
| --- | --- | --- | --- | --- | --- | --- | --- |

#### Local changes coverage

- [ ] 7e67bdba9 Fix image view flickering when image aspect ratio different from container (Imagery plugin)
- [x] 8913b87d9 Replace leftover `units` attribute by `unit` in telemetry `range` blocks processing within the tables and plots vues
- [ ] 9f2dd6220 Fix `yAxisLabel` ending up empty instead of set to the `unit` when selected range value `name` is empty
- [ ] 2eacdeb39 Make sure `yAxisLabel` update happens after the `yAxis.model.label` one at Y axis option toggle
- [ ] 67b2693a9 Change `.gl-plot-y-label` CSS format: label flows bt and symbols are rotated 90deg counter clockwise
- [ ] fc288e2d6 Add the signal `unit` to the y axis label

**Comment by nunoguedelha** [comment]
[^1]: submitted by IIT

| Status | Issue # | Pull request # | Commit # | Commit description | Remote issue | IIT[^1] | Remote pull request | IIT[^1] |
| :---: | :--- | :--- | :--- | :--- | :--- | :---: | :--- | :---: |
| | #1 | #2 | 7e67bdba9 | Fix image view flickering when image aspect ratio different from container (Imagery plugin) | | | | |
| | | | | | |  |  | |
| ✅ | #8 | #7 | 8913b87d9 | Replace leftover `units` attribute by `unit` in telemetry `range` blocks processing within the tables and plots vues | nasa/openmct/issues/3562 | nasa/openmct/issues/5540 | - | nasa/openmct/pull/5546 |
| | #8 | #7 | 9f2dd6220 | Fix `yAxisLabel` ending up empty instead of set to the `unit` when selected range value `name` is empty | - | nasa/openmct/issues/5543 | - | |
| | #8 | #7 | 2eacdeb39 | Make sure `yAxisLabel` update happens after the `yAxis.model.label` one at Y axis option toggle | - | nasa/openmct/issues/5543 | - | |
| | #8 | #7 | 67b2693a9 | Change `.gl-plot-y-label` CSS format: label flows bt and symbols are rotated 90deg counter clockwise | | | | |
| | #8 | #7 | fc288e2d6 | Add the signal `unit` to the y axis label | | | | |
| | | | | | |  |  | |
| | #9 | #10 | 6d77d3386 | Fix unit displayed in top legend not updated when we toggle Y axis key `yKey` | | | | |
| | #9 | #10 | 2bc171cd0 | Update plot-vue in-template property `unit` upon event `change:unit` instead of `change:yKey` | | | | |
| | #9 | #10 | b7f3ee516 | PlotLegendItemExpanded: replace in-template variables `unit` and `name` by computed property | | | | |
| | #9 | #10 | 20a4f9f86 | PlotLegendItemCollapsed: replace in-template variable `nameWithUnit` by computed property | | | | |
| | #9 | #10 | ab798628d | Add yKey name (label) to `PlotSeries.model` structure | | | | |
| | #9 | #10 | e6c3ccf08 | Add yKey label to plot top legend (collapsed and expanded) | | | | |
| | | | | | |  |  | |
| | #12 | #17 | 92dd3da75 | Allow duplicates of telemetry domain objects in Overlay plot | | | | |
| | | | | | |  |  | |
| | #15 | #17 | a694ae497 | Trigger Y-axis toggle on Value change in Plot Series form | | | | |
| | #15 | #17 | 2f01bdde2 | Deactivate  form (label override) | | | | |
| | #15 | #17 | 3854e7934 | Revert "Trigger Y-axis toggle on Value change in Plot Series form" | | | | |
| | #15 | #17 | 6a8bae9b2 | Add index to CompositionCollection.add | | | | |
| | #15 | #17 | 1562c022e | Fix plot-options-item for loop key | | | | |
| | #15 | #17 | c227d4317 | Insert index when adding a domain object to another domain object's composition | | | | |
| | #15 | #17 | 1579d36e4 | Add prefix and index handling to makeKeyString and parseKeyString | | | | |
| | #15 | #17 | 5584bcad3 | Fix and simplify makeKeyString and parseKeyString | | | | |
| | #15 | #17 | 3f1f28764 | Revert adding index to CompositionCollection.prototype.onProviderAdd | | | | |
| | #15 | #17 | 5d85d7902 | Revert "Add index to CompositionCollection.add" | | | | |
| | #15 | #17 | aec2060ac | Fix index handling in getPersistedSeriesConfig | | | | |
| | | | | | |  |  | |
| | #14 | #17 | 1b595e2c3 | Fixed top legend `label` and `unit` synch | | | | |
| | #14 | #17 | 648ed8e3e | Separate the Y Axis Label from the Name in the Plot Legend table | | | | |
| | | | | | |  |  | |
| | #18 | #20 | 3e76cbb9a | Fix Realtime-Telemetry-Plugin failing on Overlay Plots | | | | |

**Comment by nunoguedelha** [comment]
@traversaro @DanielePucci , I have this **Contributor License Agreement (CLA)** to sign before they can approve and merge any PRs I submit (https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md -> [https://github.com/nasa/openmct/discussions/3821](https://github.com/nasa/openmct) -> https://nasa.github.io/openmct/static/files/ind-cla-open-mct.pdf):

=> [Contributor License Agreement (CLA)](https://github.com/ami-iit/openmct/files/9095829/ind-cla-open-mct.pdf) (filled).

Please let me know if this is ok for you, or if there is any additional procedure to trigger or document to sign on IIT side. @traversaro , we can also discuss about this in tomorrow's meeting.

**Comment by traversaro** [comment]
The document is ok, feel free to send it, thanks!

**Comment by nunoguedelha** [comment]
# Summary of Contribution Process

## [Contribution Process](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md#contribution-process)

- Use topic branches named "mct<issue-number>" (all lower case, no hyphens).
- Assign the PR to a sepecific team member if you wish to draw attention (not clear if it's up to the author to do that).
- Include testing instructions or refer to those given in the issue.
- Apply same priority label as the respective issue.
- Mention "Addresses #<issue-number>" or "Closes #<issue-number>".
- When openeing issue, provide issue sverity.
    * Trivial: Minimal impact on the usefulness and functionality of the software; a "nice-to-have." Visual impact without functional impact,
    * Medium: Some impairment of use, but simple workarounds exist
    * Critical: Significant loss of functionality or impairment of use. Display of telemetry data is not affected though.
    * Blocker: Major functionality is impaired or lost, threatening mission success. Display of telemetry data is impaired or blocked by the bug, which could lead to loss of situational     * awareness.


## [Code Standards](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md#code-standards)

## [Commit Message Standards](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md#commit-message-standards)

## [Test Standards](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md#test-standards)

- Do blackbox testing. Test external behaviors, not internal details.
- Unit test specs for plugins should be defined at the plugin level. Start with one test spec per plugin named pluginSpec.js, and as this test spec grows too big, break it up into multiple test specs that logically group related tests.
- Unit tests for API or for utility functions and classes may be defined at a per-source file level.
- Do not directly invoke any private functions, but only openmct.* functions.
- Test at an appropriate level of isolation.
- you can mock openmct.telemetry.request() to feed test data to a tested view.

## Git tips
http://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project
http://git-scm.com/book/en/v2/Git-Tools-Interactive-Staging#_interactive_staging
http://git-scm.com/book/en/v2/Git-Tools-Stashing-and-Cleaning
